data/binding/json.go

```diff
@@ -0,0 +1,331 @@
+package binding
+
+import (
+	"errors"
+	"sync"
+
+	"fyne.io/fyne/v2/data/binding"
+
+	jsonvalue "github.com/Andrew-M-C/go.jsonvalue"
+)
+
+// JSONValue supports binding a jsonvalue.V{}
+type JSONValue interface {
+	binding.DataItem
+
+	GetItemString(firstParam interface{}, params ...interface{}) (binding.String, error)
+	GetItemFloat(firstParam interface{}, params ...interface{}) (binding.Float, error)
+	GetItemBool(firstParam interface{}, params ...interface{}) (binding.Bool, error)
+
+	IsEmpty() bool
+}
+
+type databoundJSON struct {
+	JSONValue
+
+	self   binding.Untyped
+	rwlock sync.RWMutex
+	source binding.String
+	last   string
+	err    error
+}
+
+// Internal type for the children accessor
+type childJSON struct {
+	source *databoundJSON
+	err    error
+
+	target []interface{}
+	first  interface{}
+}
+
+type childJSONString struct {
+	binding.String
+	generic childJSON
+}
+
+type childJSONFloat struct {
+	binding.Float
+	generic childJSON
+}
+
+type childJSONBool struct {
+	binding.Bool
+	generic childJSON
+}
+
+var (
+	errWrongType = errors.New("wrong type provided")
+)
+
+// NewJSONFromString return a data binding to a `jsonvalue.V{}` synchronized with the `String`
```

Comment:
As this is not part of the public API I think it would be better to remove it, using something like "binding to a JSON structure" I guess?
(Appears in 1 other place as well)

Reply:
Indeed.